eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -302,7 +302,10 @@ public function buildDomainFields(
         ksort($fields, SORT_NUMERIC);
 
         // Flatten array
-        return array_merge(...$fields);
+        if (count($fields) === 0) {
+            return [];
```

Comment:
This is incorrect per Content Object Model domain specification. ContentType without fields should not exist. The only thing you could do here is to throw `BadStateException`.